Source/WebCore/Modules/mediasource/SourceBuffer.cpp

```diff
@@ -2358,9 +2363,9 @@ void SourceBuffer::reportExtraMemoryAllocated()
         return;
 
     size_t extraMemoryCostDelta = extraMemoryCost - m_reportedExtraMemoryCost;
-    m_reportedExtraMemoryCost = extraMemoryCost;
 
     JSC::JSLockHolder lock(scriptExecutionContext()->vm());
+    m_reportedExtraMemoryCost = extraMemoryCost;
```

Comment:
I'm not really the author of this change so hard to tell the exact use case but checking the code suggests that indeed this is not needed.

You are right about m_reportedExtramemoryCost, it makes sense to maintain two separate variables.

NOTE that your upsteram link contains
``` 
size_t SourceBuffer::memoryCost() const
{
    return sizeof(SourceBuffer) + m_reportedExtraMemoryCost;
}
```
I believe you should return sizeof() + m_extraMemoryCost (version that can grow and shrink) - otherwise thre is no usage of this variable

Reply:
Oops! Thank you for noticing this. I've just corrected it upstream.